app/controllers/attachments_controller.rb

```
@@ -25,8 +25,9 @@ def show
   def new
     if params[:attachable_type] && params[:attachable_id]
       log_activity(@attachment.previous_changes, parent: @attachment)
-      attachable = params[:attachable_type].constantize.find(params[:attachable_id])
-      @attachment = attachable.attachments.new
+      if (attachable = set_attachable_type(params[:attachable_type]))
```

If this if condition is false then `@attachable` will be nil and that will cause problems in the view. It should probably raise an error or something I'm thinking.